pkg/node/node.go

```diff
@@ -849,6 +850,10 @@ func NewBee(interrupt chan os.Signal, addr string, publicKey *ecdsa.PublicKey, s
 		return nil, fmt.Errorf("pullsync protocol: %w", err)
 	}
 
+	depthMonitor := depthmonitor.New(kad, pullSyncProtocol, storer, stateStore, logger, warmupTime)
```

Comment:
> depthmonitor should only run in full node mode.

While I agree with that, it raises another question:  Should light and ultra-light nodes be pinning chunks into a reserve to begin with?  Or should preserveOrCache always cache and never preserve (pin) in any mode except full-node?

Reply:
@ldeffenb preserveOrCache is only called when the mode is Mode is sync or request. As light nodes and ultra light nodes would only ingest data through the API, only the PutUpload mode will be used. This will not be affecting the reserve. So this component would not affect that behavior.

Reply:
Thank you.  I guess I didn't follow that rabbit far enough down the trail.  I don't have time to look at it in detail at the moment, but aren't retrieved chunks stored with a PutRequest?  That's actually the path to the reserve that I had in mind with the original question.  If a (ultra) light node user retrieves chunks for local use, some of them will end up in the reserve via the PutRequest caching that may turn into preserving when stored locally.